### PR TITLE
feat: process check suite events

### LINKED ISSUE #629: process check suite event
**Is your feature request related to a problem? Please describe.**
Currently, `check_suite` events are not supported, we need to process `check_suite` events in order to support re-run all checks.

**Describe the solution you'd like**
Process `check_suite` events.

### PR DESCRIPTION
**Pull request by zolamk**
## Description
Process `check_suite` events.

## Related issue

Closes #629 

## Type of change

New feature (non-breaking change which adds functionality)

## How was this tested?
Unit and Manual tests.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

 
 
- [x] Ask: this pull request requires a code review before merge

### CODE REVIEW

File: handler/processors.go [hunk 1]
```diff
@@ -422,6 +422,57 @@ func processCheckRunEvent(log *logrus.Entry, token string, event *github.CheckRu
...
+
+	targetEntities := []*TargetEntity{}
+
+	eventDetails := &EventDetails{
+		EventName:   "check_suite",
+		EventAction: event.GetAction(),
+		Payload:     event,
+	}
+
+	//  if the head is from a forked repository the pull_requests array will be empty on check suite events
```
Comment (ferreiratiago) [inline handler/processors.go:436]:
```suggestion
	// When the check suite is from a head of a forked repository the pull_requests array will be empty
	// We need to fetch all the pull requests for the repository and find the one that matches the head sha
```

File: handler/processors.go [hunk 2]
```diff
@@ -422,6 +422,57 @@ func processCheckRunEvent(log *logrus.Entry, token string, event *github.CheckRu
...
+	if len(event.CheckSuite.PullRequests) == 0 {
+		prs, err := getPullRequests(token, event.GetRepo().GetFullName())
```
Comment (ferreiratiago) [inline handler/processors.go:438]: Add a log before this line.

```go
log.Infof("no pull requests found in check suite event. fetching all pull requests for repository %v", event.GetRepo().GetFullName)
```

File: handler/processors.go [hunk 3]
```diff
@@ -422,6 +422,57 @@ func processCheckRunEvent(log *logrus.Entry, token string, event *github.CheckRu
...
+		if err != nil {
+			return nil, nil, err
+		}
+
+		log.Infof("fetched %d pull requests", len(prs))
```
Comment (ferreiratiago) [inline handler/processors.go:443]:
```suggestion
		log.Infof("no pull request found with the head sha %v", event.CheckSuite.GetHeadSHA())
```

File: handler/processors.go [hunk 4]
```diff
@@ -422,6 +422,57 @@ func processCheckRunEvent(log *logrus.Entry, token string, event *github.CheckRu
...
+	}
+
+	for _, pr := range event.CheckSuite.PullRequests {
+		targetEntities = append(targetEntities, &TargetEntity{
+			Kind:   PullRequest,
+			Owner:  event.GetRepo().GetOwner().GetLogin(),
+			Repo:   event.GetRepo().GetName(),
+			Number: pr.GetNumber(),
+		})
+	}
```
Comment (ferreiratiago) [inline handler/processors.go:471]: Let's include this code inside an `else` block.

```suggestion
	} else {
		for _, pr := range event.CheckSuite.PullRequests {
			targetEntities = append(targetEntities, &TargetEntity{
				Kind:   PullRequest,
				Owner:  event.GetRepo().GetOwner().GetLogin(),
				Repo:   event.GetRepo().GetName(),
				Number: pr.GetNumber(),
			})
		}
	}
```

**Review by ferreiratiago — CHANGES_REQUESTED** [review]

File: handler/processors.go (hunk 3 above)
Comment (zolamk) [reply handler/processors.go:443]: I think some prs might still be found at this point right

Comment (ferreiratiago) [reply handler/processors.go:446]: I'm sorry. This comment was to be done on line [462](https://github.com/reviewpad/reviewpad/pull/630/files#diff-d07de10e776881f04ef3111a10250bfc72303ef3f0c71ffdd259c95632cbdfa3R462).

**Review by ferreiratiago — APPROVED** [review]

**Pull request merged**